experiments/AMIP/coupler_driver.jl

```diff
@@ -554,38 +619,43 @@ else
+
+# 3. surface density (`ρ_sfc`): calculated by the coupler by adiabatically extrapolating atmospheric thermal state to the surface.
+# For this, we need to import surface and atmospheric fields. The model sims are then updated with the new surface density.
 import_combined_surface_fields!(cs.fields, cs.model_sims, cs.boundary_space, turbulent_fluxes)
 import_atmos_fields!(cs.fields, cs.model_sims, cs.boundary_space, turbulent_fluxes)
 update_model_sims!(cs.model_sims, cs.fields, turbulent_fluxes)
 
-# 2) each surface component model calculates its own vapor specific humidity (q_sfc)
-# TODO: the q_sfc calculation follows the design of the bucket q_sfc, but it would be neater to abstract this from step!
+# 4. surface vapor specific humidity (`q_sfc`): step surface models with the new surface density to calculate their respective `q_sfc` internally
+## TODO: the q_sfc calculation follows the design of the bucket q_sfc, but it would be neater to abstract this from step! (#331)
 step!(land_sim, Δt_cpl)
 step!(ocean_sim, Δt_cpl)
 step!(ice_sim, Δt_cpl)
 
-# 3) coupler re-imports updated surface fields and calculates turbulent fluxes, while updating atmos sfc_conditions
+# 5. turbulent fluxes: now we have all information needed for calculating the initial turbulent surface fluxes using the combined state
+# or the partitioned state method
 if turbulent_fluxes isa CombinedStateFluxes
-    # calculate fluxes using combined surface states on the atmos grid
+    ## import the new surface properties into the coupler (note the atmos state was also imported in step 3.)
     import_combined_surface_fields!(cs.fields, cs.model_sims, cs.boundary_space, turbulent_fluxes) # i.e. T_sfc, albedo, z0, beta, q_sfc
+    ## calculate turbulent fluxes in surface models and save the weighted average in coupler fields
     combined_turbulent_fluxes!(cs.model_sims, cs.fields, turbulent_fluxes) # this updates the atmos thermo state, sfc_ts
 elseif turbulent_fluxes isa PartitionedStateFluxes
-    # calculate turbulent fluxes in surface models and save the weighted average in coupler fields
+    ## calculate turbulent fluxes in surface models and save the weighted average in coupler fields
     partitioned_turbulent_fluxes!(cs.model_sims, cs.fields, cs.boundary_space, MoninObukhovScheme(), thermo_params)
 
-    # update atmos sfc_conditions for surface temperature
-    # TODO: this is hard coded and needs to be simplified (need CA modification)
+    ## update atmos sfc_conditions for surface temperature
+    ## TODO: this is hard coded and needs to be simplified (req. CA modification) (#479)
     new_p = get_new_cache(atmos_sim, cs.fields)
-    CA.SurfaceConditions.update_surface_conditions!(atmos_sim.integrator.u, new_p, atmos_sim.integrator.t) # sets T_sfc (but SF calculation not necessary - CA)
+    CA.SurfaceConditions.update_surface_conditions!(atmos_sim.integrator.u, new_p, atmos_sim.integrator.t) ## sets T_sfc (but SF calculation not necessary - CA)
```

Comment:
what does the comment mean - "SF calculation not necessary" ?

Reply:
This is a comment that the CA interface requires an SF calculation to update its surface conditions, but this should not be necessary. In other words, the setting of surface conditions should be split from calculating fluxes in in CA. I clarified the comment.